File: app/src/main/java/podtest/jiki/se/podtest/start/EpisodeViewHolder.java
package podtest.jiki.se.podtest.start;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import butterknife.BindView;
import butterknife.ButterKnife;
import podtest.jiki.se.podtest.R;

public class EpisodeViewHolder extends RecyclerView.ViewHolder implements RowItemContract.EpisodeView {
    public View container;

    @BindView(R.id.first_line) TextView firstRow;
    @BindView(R.id.second_line) TextView secondRow;
    @BindView(R.id.left_image) ImageView leftImage;
    @BindView(R.id.right_image) ImageView rightImage;

    private RowItemContract.EpisodePresenter presenter;
    private Context context;

    public EpisodeViewHolder(View view, Context context) {
        super(view);

        this.context = context;
        this.container = view;

        ButterKnife.bind(this, container);
        container.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (presenter != null) {
                    presenter.itemClicked(container.getTag(), RowItemContract.Source.CONTAINER);
                }
            }
        });
        rightImage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (presenter != null) {
                    presenter.itemClicked(container.getTag(), RowItemContract.Source.RIGHT_ICON);
                }
            }
        });
    }

    @Override
    public void setFirstRow(String text) {
        firstRow.setText(text);
    }

    @Override
    public void setSecondRow(String text) {
        secondRow.setText(text);
    }

    @Override
    public void setThumbnail(String url) {
        Glide.with(context)
                .load(url)
                .into(leftImage);
    }

    @Override
    public void setRightIcon(int drawableResourceId) {
        rightImage.setImageResource(drawableResourceId);
    }

    @Override
    public void setTag(Object tag) {
        container.setTag(tag);
    }

    @Override
    public void setPresenter(RowItemContract.EpisodePresenter presenter) {
        this.presenter = presenter;
    }
}

File: app/src/main/java/se/asapehrsson/podtest/EpisodeViewHolder.kt
package se.asapehrsson.podtest

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.bumptech.glide.Glide

class EpisodeViewHolder(val container: android.view.View, private val context: Context) : EpisodeContract.View, RecyclerView.ViewHolder(container) {

    @BindView(R.id.icon_container) lateinit var iconContainer: View
    @BindView(R.id.episode_image) lateinit var episodeImage: ImageView
    @BindView(R.id.first_line) lateinit var firstRow: TextView
    @BindView(R.id.second_line) lateinit var secondRow: TextView

    private var presenter: EpisodeContract.Presenter? = null

    init {
        ButterKnife.bind(this, container)

        container.setOnClickListener {
            container.tag?.let { handleClick(EpisodeContract.Request.PLAY_EPISODE) }
        }

        iconContainer.setOnClickListener {
            container.tag?.let { handleClick(EpisodeContract.Request.SHOW_DETAILS) }
        }
    }

    private fun handleClick(request: EpisodeContract.Request) {
        presenter?.itemClicked(container.tag, request)
    }

    override fun setFirstRow(text: String?) {
        firstRow.text = text
    }

    override fun setSecondRow(text: String?) {
        secondRow.text = text
    }

    override fun setThumbnail(url: String?) {
        Glide.with(context)
                .load(url)
                .into(episodeImage)
    }

    override fun setTag(tag: Any) {
        container.tag = tag
    }

    override fun setPresenter(presenter: EpisodeContract.Presenter) {
        this.presenter = presenter
    }
}
